Add initial features for torchvision transforms

grmvvr opened the pull request:
-Currently supports torchvision transforms for PIL Image and
torch.*Tensor.
-Conversion transforms to convert PIL Image or np.ndarray to torch
Tensor and the other way around.

grmvvr committed:
Add initial features for torchvision transforms

-Currently supports torchvision transforms for PIL Image and
torch.*Tensor.
-Conversion transforms to convert PIL Image or np.ndarray to torch
Tensor and the other way around.

faustomilletari commented on eisen_extras/torchvision/transforms.py:
```diff
@@ -0,0 +1,214 @@
+
+class CenterCrop(PILImageTorchvisionTransforms):
+    """# TODO"""
+    def __init__(self, fields: List[str], *args, **kwargs):
```
This is good. it probably already works fine (this holds for all of the transforms here). We will need some tests and docstrings.

grmvvr replied:
Yes, the docstrings and test cases are still missing here. Will implement those  next.

faustomilletari commented on eisen_extras/torchvision/transforms.py:
```diff
@@ -0,0 +1,214 @@
+
+    def _check_data_type(self, instance):
+        """# TODO"""
+        if not isinstance(instance, Image):
```
I am unsure if all the expected types for torchvision transforms are PIL Image. torchvision transforms might require some other types. One thing one could do is to have the type defined during init

__init__(self):
    self.expected_type = Image

_check_data_type(self, instance):
   ...
   if not isinstance(instance, self.expected_type)
   ...

grmvvr replied:
Good point. Not sure if you saw that there is also another parent class below for Tensor input. We have two parent classes here for transforms which need PIL Image input and another class for torch.*Tensor input. Do you mean here we let the user input the data type during init of the transform or  do we use a single parent class for all the transforms? :)

faustomilletari replied:
I think it's fine in the way you are doing it :) I didn't see there were two base classes

faustomilletari replied:
we can also store the expected type during init but without asking the user to supply it. it would belong to the class and actually be hardcoded in it.

grmvvr replied:
Sounds good, I'll include the expected_type variable then.

grmvvr committed:
Add expected_type variable to base classes.

grmvvr committed:
Add docstring to all transforms

eisen-ai commented:
hello! thanks! is this close to be ready to merge?

grmvvr commented:
Hi, I just need to finish the test cases. Then it should be ready.

grmvvr committed:
Update Random Transforms to inherit from Torchvision's Random Transforms

-Individual transforms from list of transforms should handle which
keynames the transform will be applied to.

grmvvr committed:
Add RandomOrder transform from Torchvision.

grmvvr committed:
Add Compose and Lambda transforms

grmvvr committed:
Add test cases for PIL Image transforms and Random transforms.

grmvvr committed:
Update docstring of PIL Image and Tensor transforms

- Include an explicit example of input data

grmvvr committed:
Add test cases for torch.*Tensor transforms

grmvvr commented:
Hi @eisen-ai @faustomilletari, the test cases are done and I think these wrapped transforms should be usable. Please have a look when you have time and let me know what you think.
